pkg/frontend/openshiftcluster_putorpatch_test.go

```diff
@@ -942,3 +943,375 @@ func TestPutOrPatchOpenShiftCluster(t *testing.T) {
 		})
 	}
 }
+
+// TODO(mjudeiki): This is shameful copy of previous api test. It should be extended
+// when new features lands for new api. and merged with the test above when new
+// api becomes default
+func TestPutOrPatchOpenShiftClusterV20210131preview(t *testing.T) {
```

Comment:
Don't we need to update `TestPutOrPatchOpenShiftClusterV20200430` to also test for system data? I believe all API versions now handle this.

I would probably remove this test and keep only one test (`TestPutOrPatchOpenShiftCluster`) and update it with cases to handle system data. And deal with handling multiple API versions later.

Reply:
> I believe all API versions now handle this.

This is wrong: this PR updates only one API version. Still I'm inclined to hae only one test (at least for now): make `stompSystemData` mockable, test that we call it and test `stompSystemData` separately. as suggested in https://github.com/Azure/ARO-RP/pull/1412#discussion_r619141869.

Reply:
To add to older api would mean changing api spec. And this is not allowed as this would be considered breaking change :/
So we are not retrofitting this into older apis. Old apis does not have converters into external model so we will have data in the database, but not in the external model. 

The problem with having one tests is that all the checks relies on external API model. So having full tests required support for multiple apis in the same tests. And I think it might be bit overcomplication for the test itself. Considering old API should not change we should leave the tests and not change it too. 
Or you think otherwise?

Reply:
Im still bit on the fence. For few reasons
Mocks are for interfaces. And I dont think this function qualifies to have its own interface just for this.
We gonna have more api specific tests with new api so test specific putOrPatch will be needed either way.
Having separate function will help us to make sure we dont break old apis when we move all our code to new ones (apis are not deprecatable in azure for lifetime of service).

So all this looks like a lot of effort which I dont think is required here now.

Reply:
> Mocks are for interfaces. And I dont think this function qualifies to have its own interface just for this.

It definitevely doesn't deserve own interface. By mock I mean dummy/fake implementation here. I suggest that we do something similar to how we mock `time.Now()`. Here is an example ([code](https://github.com/Azure/ARO-RP/blob/1cc637ac2fb8e4707fa3f11b88f4ffaa2178fbc8/pkg/util/clientauthorizer/arm.go#L36) and [test](https://github.com/Azure/ARO-RP/blob/1cc637ac2fb8e4707fa3f11b88f4ffaa2178fbc8/pkg/util/clientauthorizer/arm_test.go)). In the same file you can see how we mock HTTP client's `Do` call.

So we in the test we will be replacing `stompSystemData` with a mock function and we will just need to check that we call it and that's it.


> We gonna have more api specific tests with new api so test specific putOrPatch will be needed either way.
> Having separate function will help us to make sure we dont break old apis when we move all our code to new ones (apis are not deprecatable in azure for lifetime of service).

We need something better than having N separate tests for each API version. Let's deal with it later. Properly.